### DISCUSSION TITLE
Add `isNullish` function

### DISCUSSION THREAD

**Discussion by coldsmirk** [radashi-org/radashi#274] [Ideas]
Currently, there is no function in the library to determine whether a value is nil (`undefined` or `null`).

```ts
/**
 * Check if the value is null or undefined
 *
 * @param value - The value to check
 * @returns {boolean} - True if the value is null or undefined, false otherwise
 */
export function isNullish(value: unknown): value is null | undefined {
  return value === null || value === undefined;
}
```

**Comment by aleclarson** [comment]
Hey @ilxqx, thanks for the suggestion! (and welcome to the Radashi community ✌️)

I'd be open to adding this, even though we generally avoid syntax sugar.

- **Advantages**
  - Avoid mistyping `== null` as `=== null`
  - Possibly easier to understand for junior devs
- **Disadvantages**
  - Could be used as justification to add more syntax sugar
  - “Less typing” or ”less typos” are not goals of Radashi

It would be named `isNullish` though, to align with common parlance.

What do you think, @radashi-org/core?

**Reply by coldsmirk** [reply]
The name `isNullish` is also ok. The reason I chose `isNil` is that I borrowed from the conventions in Golang.😄

**Reply by aleclarson** [reply]
Lodash also uses `isNil` but I think `isNullish` is more understandable in the context of JavaScript.

Looks like @MarlonPassos-git marked this idea as "accepted", so anyone is free to submit a PR.

**Reply by coldsmirk** [reply]
@aleclarson Hi, I just submitted a Pull Request regarding this function.
https://github.com/radashi-org/radashi/pull/277
